### PR TITLE
Issue 4 remove paladin on leavegrp

### LINKED ISSUE #4: Remove paladin that left the party/raid from pallypower
Hi!

A the moment, when a paladin leave the group, PallyPower doesn't removes it from the UI, event with the "update" button (not sure about the name in english) and one needs to /reload to actually cleanup the pal list.

Would be great if pallypower could automatically remove them when they leave, or at least update the list when using the update button from the blessing assignment window

### PR DESCRIPTION
**Pull request by SaschaJankowiak**
FIXES #4 

The old logic used active scanning of ChatMessages to determine if a user left or not. This could have only worked for english clients as the strings to parse were hardcoded. Changed the behavior to do a Pally-Rescan on Group-Leave

### CODE REVIEW

File: PallyPowerValues.lua [hunk 1]
```diff
@@ -429,9 +429,12 @@ PallyPower.Auras = PallyPower.isWrath and {
 	[8] = GetSpellInfo(32223),	-- Crusader Aura
 }
 
-PallyPower.Cooldowns = {
-	[1] = {633, 2800, 10310, 27154},	-- Lay On Hands
-	[2] = {19752},						-- Divine Intervention
+PallyPower.Cooldowns = PallyPower.isWrath and {
+	[1] = {633, 2800, 10310, 27154, 48788},	-- Lay On Hands
```
Comment (Treeston) [inline PallyPowerValues.lua:433]: make a separate pull request for wrath changes

File: PallyPower.lua [hunk 2]
```diff
@@ -1613,42 +1614,48 @@ function PallyPower:GROUP_LEFT(event)
...
-							self:ScanCooldowns()
-							self:ScanInventory()
-							self:SendSelf()
-							self:SendMessage("REQ")
-							self:UpdateLayout()
-							self:UpdateRoster()
-						end
-					)
+		end
+		if found < countAllPallys then -- Zid: if AllPallys count is reduced do a fresh setup
```
[the remark this answers is not in the document]
Comment (SaschaJankowiak) [reply PallyPower.lua]: bad refactoring - the scope of the loop should not include that

File: PallyPower.lua [hunk 3]
```diff
@@ -1613,43 +1613,46 @@ function PallyPower:GROUP_LEFT(event)
...
+	local units
+	local current_units = {}
+	if IsInRaid() then
+		units = raid_units
+	else
+		units = party_units
+	end
+	for _, unitid in pairs(units) do
+		if unitid and UnitExists(unitid) then
+			tinsert(current_units, GetUnitName(unitid, true))
```
Comment (Treeston) [inline PallyPower.lua:1629]: what's the point of making this `current_units` table in the first place?

just check `AllPallys[GetUnitName(...)]` here, and do the count

ps check that the unit is not a pet to make sure there isn't any funny business if pet names share player names

File: PallyPower.lua [hunk 4]
```diff
@@ -1613,43 +1613,46 @@ function PallyPower:GROUP_LEFT(event)
...
+	local countAllPallys = 0
+	for _ in pairs(AllPallys) do countAllPallys = countAllPallys + 1 end
+	for _, unitname in pairs(current_units) do
+		if AllPallys[unitname] then
+			found = found + 1 
+		end
+	end
+	if found < countAllPallys then -- Zid: if AllPallys count is reduced do a fresh setup
+		C_Timer.After(
+			2.0,
```
Comment (Treeston) [inline PallyPower.lua:1642]: isn't this going to end up scheduled and executed twice, if players are leaving/joining repeatedly? what's the point of having the 2s delay here?

Comment (Treeston) [reply PallyPower.lua:1642]: change the delay to 0.5 then

File: PallyPower.lua [hunk 5]
```diff
@@ -1613,43 +1613,46 @@ function PallyPower:GROUP_LEFT(event)
...
+		units = raid_units
+	else
+		units = party_units
+	end
+	for _, unitid in pairs(units) do
+		if unitid and UnitExists(unitid) then
+			tinsert(current_units, GetUnitName(unitid, true))
+		end
+	end
+	local found = 0
```
Comment (Treeston) [inline PallyPower.lua:1632]:
```lua
local found = 0
for _, unitid in pairs(units) do
    if unitid and (not unitid:find("pet")) and UnitExists(unitid) then
        if AllPallys[GetUnitName(unitid, true)] then found = found+1 end
    end
end
```

**Review by Treeston — APPROVED** [review]
